src/Service.php

```diff
@@ -74,28 +82,46 @@ class Service
     /**
      * Client constructor.
      * @param string $apiSecret
-     * @param string $measurementId
+     * @param string|null $measurementId
      */
-    public function __construct(string $apiSecret, string $measurementId)
+    public function __construct(string $apiSecret, ?string $measurementId = null)
     {
         $this->setApiSecret($apiSecret);
-        $this->setMeasurementId($measurementId);
+        if ($measurementId) {
+          @trigger_error('Creating a measurement service instance with a measurement ID passed to the constructor is deprecated in v0.1.3 and removed in v0.2.0. Use ::setMeasurementId() or ::setFirebaseId() directly, instead.', E_USER_DEPRECATED);
+          $this->setMeasurementId($measurementId);
+        }
     }
 
     /**
      * @param AbstractRequest $request
+     * @param bool $debug
      * @return BaseResponse
      * @throws Exception\ValidationException
      * @throws Exception\HydrationException
      */
-    public function send(AbstractRequest $request)
+    public function send(AbstractRequest $request, bool $debug = false)
     {
-        $request->validate();
-        $response = $this->getHttpClient()->post($this->getEndpoint(), $request->export(), $this->getOptions());
+        $request->validate($this->measurementId ? 'web' : 'firebase');
+        $response = $this->getHttpClient()->post($this->getEndpoint($debug), $request->export(), $this->getOptions());
 
-        return new BaseResponse($response);
+        return !$debug
+            ? new BaseResponse($response)
+            : new DebugResponse($response);
+    }
+    
+    /**
+     * @param AbstractRequest $request
+     * @return BaseResponse
+     * @throws Exception\ValidationException
+     * @throws Exception\HydrationException
+     * @deprecated Use ::send() with $debug = true, instead.
+     */
+    public function sendDebug(AbstractRequest $request)
```

Comment:
Hi @bradjones1, could you please share your thoughts on why should we make this deprecated?

Reply:
Sure, I understand that part. I just wonder why make sendDebug deprecated and not leave it in current form (calling method send with $debug = true) without deprecation warning.

Reply:
Yeah I mean I don't have a strong feeling about it. My thought process is that having a method that only serves to add a parameter and invoke another method is not particularly DRY, but it could also be seen as a convenience method. But certainly not a huge deal to me.